dsl/src/main/java/com/powsybl/dsl/ast/ExpressionPrinter.java

```diff
@@ -37,102 +33,110 @@ public static void print(ExpressionNode node, PrintStream out) {
         node.accept(new ExpressionPrinter(out), null);
     }
 
+    private static void write(ExpressionNode node, StringWriter out) {
+        node.accept(new ExpressionPrinter(out), null);
+    }
+
     public ExpressionPrinter(PrintStream out) {
-        this.out = Objects.requireNonNull(out);
+        this.out = new PrintWriter(Objects.requireNonNull(out));
+    }
+
+    public ExpressionPrinter(Writer writer) {
+        out = new PrintWriter(Objects.requireNonNull(writer));
     }
```

Comment:
Looking at the javadoc it's not clear to me why we could not have a constructor that takes an OutputStream instead of a PrintStream. That will make `ExpressionPrinter` and `ActionExpressionPrinter` more convenient to use as we just wrap the objects in a `PrintWriter` that seems to work well with a simple OutputStream.

Reply:
Agreed. The only thing which is not completely clear now is that this constructor will use "default charset" to write to the output stream: I think we should mention it in javadoc of the constructor.